From 60850e455a694b0ae9ee71493b0319af1477f56d Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Wed, 27 May 2020 07:14:55 -0500
Subject: [PATCH 33/39] arm64: dts: beacon-imx8mm: Fix USDHC2 CD Pin

The card detetect pin is currently setup to use the default pin,
however, it doesn't appear to be recognizing when the card is
either inserted or removed when booting from devices other than
USDHC2 (uSD card on baseboard).

This patch changes the USDCH2 CD to use GPIO, and configures the
pinmux for the for SD2_CD_B to GPIO2_IO12.

Fixes: PHOENIX8-158

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
index ef0f2323ea29..f7d875743957 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
@@ -281,7 +281,7 @@
 
 	pinctrl_usdhc2_gpio: usdhc2grpgpio {
 		fsl,pins = <
-			MX8MM_IOMUXC_SD2_CD_B_USDHC2_CD_B	0x41
+			MX8MM_IOMUXC_SD2_CD_B_GPIO2_IO12	0x159
 			MX8MM_IOMUXC_SD2_RESET_B_GPIO2_IO19	0x41
 		>;
 	};
@@ -549,6 +549,7 @@
 	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
 	pinctrl-2 = <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
 	bus-width = <4>;
+	cd-gpios = <&gpio2 12 GPIO_ACTIVE_LOW>;
 	vmmc-supply = <&reg_usdhc2_vmmc>;
 	status = "okay";
 };
-- 
2.17.1

